packages/modules/product/src/services/product-module-service.ts

```diff
@@ -1339,6 +1356,8 @@ export default class ProductModuleService<
       productData.discountable = false
     }
 
+    this.validateProductHandle(product)
```

Comment:
I think it's better to call the `validate*` outside of `normalize*`, WDYT? Alternatively we call validate at the end of the function, not in the middle of it. Also, maybe we can name it more generically, eg. `validateCreateProduct` and we will add more validations in it as we go.

Reply:
Yeah, it indeed makes sense to keep validation outside of normalization. So I am going to move it to `create_` and `update_` methods.

> Also, maybe we can name it more generically, eg. validateCreateProduct and we will add more validations in it as we go.

Should we keep it even more generic like `validateProductPayload`? Because (atleast as of now) we are validating handle for both the actions.

Reply:
@thetutlage yeah makes sense 👍